webapp/src/Components/ReplaysSavedGroup/GroupAddDialog.tsx

```diff
@@ -0,0 +1,112 @@
+import {Dialog, DialogActions, DialogContent, DialogTitle, Typography} from "@material-ui/core"
+import Button from "@material-ui/core/Button"
+import Grid from "@material-ui/core/Grid"
+import * as React from "react"
+import {MatchHistoryResponse, Replay, ReplaysSearchQueryParams} from "../../Models"
+import {addGames, searchReplays} from "../../Requests/Replay"
+import {ReplaysSearchWithQueryString} from "../ReplaysSearch/Filter/ReplaysSearchWithQueryString"
+import {ReplaysSearchResultDisplay} from "../ReplaysSearch/ReplaysSearchResultDisplay"
+import {WithNotifications, withNotifications} from "../Shared/Notification/NotificationUtils"
+
+interface OwnProps {
+    openDialog: boolean
+    onCloseDialog: () => void
+    group: string
+}
+
+type Props = OwnProps & WithNotifications
+
+interface State {
+    replaySearchResult?: MatchHistoryResponse
+    queryParams?: ReplaysSearchQueryParams
+    page: number
+    limit: number
+}
+
+class GroupDialogComponent extends React.Component<Props, State> {
+    constructor(props: Props) {
+        super(props)
+        this.state = {page: 0, limit: 10}
+    }
+
+    public componentDidUpdate(prevProps: unknown, prevState: Readonly<State>) {
```

Comment:
prevProps isn't type Props? (usually it prompts to autocreate a method with the right types if you ctrl-alt-o when your cursor is in a component.)